rotkehlchen/tests/unit/decoders/test_paraswap.py

```diff
@@ -0,0 +1,761 @@
+import pytest
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.ethereum.modules.paraswap.constants import (
+    PARASWAP_AUGUSTUS_ROUTER,
+    PARASWAP_TOKEN_TRANSFER_PROXY,
+)
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.chain.evm.decoding.paraswap.constants import CPT_PARASWAP
+from rotkehlchen.constants.assets import A_ENS, A_ETH, A_LINK, A_SUSHI, A_USDC, A_USDT, A_WBTC
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.evm_event import EvmEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.tests.utils.ethereum import get_decoded_events_of_transaction
+from rotkehlchen.types import Location, TimestampMS, deserialize_evm_tx_hash
+
+A_ROUTE = Asset('eip155:1/erc20:0x16ECCfDbb4eE1A85A33f3A9B21175Cd7Ae753dB4')
+A_stETH = Asset('eip155:1/erc20:0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84')
+A_MPL = Asset('eip155:1/erc20:0x33349B282065b0284d756F0577FB39c158F935e6')
+A_TKST = Asset('eip155:1/erc20:0x7CdBfC86A0BFa20F133748B0CF5cEa5b787b182c')
+A_PRISMA = Asset('eip155:1/erc20:0xdA47862a83dac0c112BA89c6abC2159b95afd71C')
+A_UDOODLE = Asset('eip155:1/erc20:0xa07dCC1aBFe20D29D87a32E2bA89876145DAfB0a')
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [['0xf8fa0bB0798489445577fA7387dcB2125C361c28']])
+def test_simple_swap_no_fee(database, ethereum_inquirer, ethereum_accounts):
+    tx_hash = deserialize_evm_tx_hash(
+        '0xeab04bca5abb6a794fd83e3c336b128824a2d3a72abf565d82a6ecd39d5929ef',
+    )
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    user_address = ethereum_accounts[0]
+    timestamp = TimestampMS(1704814775000)
+    swap_amount = '3860.977240111926214656'
+    received_amount = '3997.851328'
+    approval_amount = '115792089237316195423570985000000000000000000000000000000000'
+    gas_fees = '0.003865522700588692'
```

Comment:
Here and in other tests to save lines do

```suggestion
    swap_amount, received_amount, approval_amount, gas_fees  = '3860.977240111926214656',  '3997.851328', 115792089237316195423570985000000000000000000000000000000000', '0.003865522700588692'  # noqa: E501
```